api/v1beta1/common_types.go

```diff
@@ -48,3 +48,35 @@ type CinderDebug struct {
 	// Service enable debug
 	Service bool `json:"service,omitempty"`
 }
+
+// CinderCephBackend defines the Ceph client parameters
+type CinderCephBackend struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClusterFSID"
+	// CephClusterFSID defines the fsid
+	CephClusterFSID string `json:"cephFsid,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClusterMonHosts"
+	// CephClusterMons defines the commma separated mon list
+	CephClusterMonHosts string `json:"cephMons,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClientKey"
+	// CephClientKey set the Ceph cluster key used by Cinder
+	CephClientKey string `json:"cephClientKey,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephUser"
+	// CephUser set the Ceph cluster pool used by Cinder
+	CephUser string `json:"cephUser,omitempty"`
+	// +kubebuilder:validation:Optional
+	// CephPools - Map of chosen names to spec definitions for the Ceph cluster
+	// pools used by Cinder
+	CephPools map[string]CephPoolSpec `json:"cephPools,omitempty"`
+}
```

Comment:
Just curious as to whether these should be required if the user has chosen to define/include a `CephBackend` in the `CinderSpec`.  If the user has included a `CephBackend` in the main spec, is it valid to set some of these and not set others?  Or did you perhaps make it this way because the k8s API validator was complaining without everything being optional?  I think I recall seeing k8s validation not allow unset sub-fields even if the containing field itself was optional and the user had not included it in the spec.

Reply:
Yeah, thanks, this is a good point.
We require at least to set `CephClusterFSID`,  `CephClusterMonHosts` and the `CephClientKey`. These are the minimum amount of info we need to start the Ceph backend.
In general, if `CephUser` and `CephPools` are not specified, we have defaults in place, which means  we can keep them as "Optional".
Now, being that said,  I think we can change `kubebuilder:validation:Optional` to `kubebuilder:validation:Required`, what do you think?

Reply:
Sure, that seems reasonable.  Also, when we add that annotation, we should remove `omitempty`.  It's weird that there are sort of two ways to declare "optional", but it is what it is I suppose.  

I do wonder though if we'll run into the validation concern I mentioned (about required sub-fields in an optional containing field).  I guess we'll find out. :)

Reply:
Yeah, we should remove both `omitempty` and the `default`, otherwise from a quick test I see the k8s API validator set the that value to the field even if not provided!
Removing both `omitempty` and the `+kubebuilder:default=<Value>` block the user during the object creation.

E.g.
```
> oc create -f out/openstack/cinder/cr/cinder_v1beta1_cinder.yaml
The Cinder "cinder" is invalid: spec.cephBackend.cephClientKey: Required value
```